covins_backend/src/covins_backend/RelNonCentralPosSolver.cpp

```diff
@@ -0,0 +1,392 @@
+// Header file
+#include "covins_backend/RelNonCentralPosSolver.hpp"
+
+// Standard includes
+#include <cmath>
+#include <vector>
+
+#include <eigen3/Eigen/Core>
+
+// CoVINS
+#include "covins_backend/keyframe_be.hpp"
+#include "covins_base/config_backend.hpp"
+
+// opengv related includes
+#include <opengv/sac/Ransac.hpp>
+#include <opengv/sac_problems/relative_pose/NoncentralRelativePoseSacProblem.hpp>
+#include "matcher/opengv/rel_pose/frame-relative-adapter.hpp"
+#include "matcher/opengv/rel_pose/FrameNoncentralRelativeAdapter.hpp"
+#include "matcher/opengv/sac_problems/frame-relative-pose-sac-problem.hpp"
+#include "opengv/relative_pose/methods.hpp"
+
+#include <opencv2/features2d.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/core/eigen.hpp>
+
+
+
+namespace covins {
+
+// The class constructor
+RelNonCentralPosSolver::RelNonCentralPosSolver(const size_t minInliers, const double ransacProb,
+                           const size_t maxIter)
+    : mMinInliers(minInliers),
+    mRansacProb(ransacProb),
+    mMaxIter(maxIter)
+{
+  mMinInliers_17PT = covins_params::placerec::nc_rel_pose::min_inliers;
+  mMaxIter_17PT = covins_params::placerec::nc_rel_pose::max_iters;
+  mImgMatches = covins_params::placerec::rel_pose::min_img_matches;
+  mMatchThreshold = covins_params::features::img_match_thres;
+  mMinInliers = covins_params::placerec::rel_pose::min_inliers;
+  mMaxIter = covins_params::placerec::rel_pose::max_iters;
+  mRansacProb = covins_params::placerec::ransac::probability;
+  mCov_iter = covins_params::placerec::nc_rel_pose::cov_iters;
+  mCov_max_iter = covins_params::placerec::nc_rel_pose::cov_max_iters;
+  mMax_cov = covins_params::placerec::nc_rel_pose::cov_thres;
+  mRP_err = covins_params::placerec::nc_rel_pose::rp_error_cov;
+  double thres_17pt = covins_params::placerec::nc_rel_pose::rp_error;
+  mThres_17PT = 2.0 * (1.0 - cos(atan(thres_17pt * 1 / 800.0)));
+  
+}
+
+// Set the parameters used for RANSAC
+void RelNonCentralPosSolver::setRansacParams(const int minInliers,
+    const double ransacProb, const int maxIter) {
+  mMinInliers = minInliers;
+  mRansacProb = ransacProb;
+  mMaxIter = maxIter;
+}
+
+bool RelNonCentralPosSolver::computeNonCentralRelPose(
+    const KeyframePtr QKF, const KeyframePtr CKF,
+    const double threshold, Eigen::Matrix4d &Tc1c2, Eigen::Matrix<double, 6, 6> &cov_loop) {
+
+  // Find matches between CKF and QKF
+  // std::cout << "Attempting NON-Relative Pose Solver" << std::endl;
+
+  // 3 v 2 Matching of Multicamera Systems (See publication for reference)
+  size_t n_ckfs = 3;
+  size_t n_qkfs = 2;
+  TransformType T_QKF_cw = QKF->GetPoseTcw();
+  TransformType T_CKF_cw = CKF->GetPoseTcw();
+  TransformType T_init;
+  cov_loop = Eigen::Matrix<double,6,6>::Identity();
+  // Initialize the adapter vectors
+  KeyframeVector view_A;
+  KeyframeVector view_B;
+  std::vector<std::vector<Matches>> match_vect(n_qkfs);
+  std::vector<std::vector<TransformType>> TF_vect(2);
```

Comment:
No this will always be 2 irrespective of the number of qkfs since the TF_vect[0] and TF_vect[1] consists of a vector of transformations for QKF and CKF respectively

Reply:
Ah ok, then I misinterpreted this.